packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart

```diff
@@ -451,15 +449,16 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface
     final result = await _cognitoIdp.resendConfirmationCode(
       cognito.ResendConfirmationCodeRequest.build((b) {
         b
-          ..clientId = _userPoolConfig.appClientId
+          ..clientId = _authOutputs.userPoolClientId
           ..username = username
           ..analyticsMetadata = _analyticsMetadata?.toBuilder();
 
-        final clientSecret = _userPoolConfig.appClientSecret;
+        // ignore: invalid_use_of_internal_member
+        final clientSecret = _authOutputs.appClientSecret;
         if (clientSecret != null) {
           b.secretHash = computeSecretHash(
             username,
-            _userPoolConfig.appClientId,
+            _authOutputs.userPoolClientId!,
```

Comment:
Q: do we need a proper null check before force unwrapping `_authOutputs.userPoolClientId!`?

Reply:
we have a check when we get the _authOutputs from the dependency manager. it is line128